elasticdl/python/tests/optimizer_wrapper_test.py

```diff
@@ -14,6 +21,143 @@
 from elasticdl.python.tests.mock_kv_store import MockKvStore
 
 
+def _prepare_random_data(
+    iters_per_epoch,
+    batch_size,
+    input_length,
+    input_dim,
+    is_sparse,
+    random_seed,
+):
+    """
+    Generate data for training. `is_sparse=False` means we require that all
+    embedding ids appear in every batch data.
+    """
+    random.seed(random_seed)
+    X, Y = [], []
+    if not is_sparse:
+        assert input_length > input_dim, "`input_length` should be larger "
+        "than `input_dim` when dense data required."
+
+    def _gen_single_data(choices, input_length, is_sparse):
+        if not is_sparse:
+            data = copy.copy(choices)
+        else:
+            data = []
+        while len(data) < input_length:
+            data.append(random.choice(choices))
+        random.shuffle(data)
+        return data
+
+    for i in range(iters_per_epoch):
+        f1_batch, f2_batch, f3_batch, y_batch = [], [], [], []
+        if is_sparse:
+            choices = random.choices(range(input_dim), k=input_dim // 2)
+        else:
+            choices = list(range(input_dim))
+        for j in range(batch_size):
+            f1_batch.append(_gen_single_data(choices, input_length, is_sparse))
+            f2_batch.append(_gen_single_data(choices, input_length, is_sparse))
+            f3_batch.append(_gen_single_data(choices, input_length, is_sparse))
+            y_batch.append(random.randint(0, 1))
+        X.append(
+            {
+                "f1": np.array(f1_batch),
+                "f2": np.array(f2_batch),
+                "f3": np.array(f3_batch),
+            }
+        )
+        Y.append(y_batch)
+    return X, Y
+
+
+def _train(model, optimizer, X, Y, loss_fn, random_seed):
+    """Train model with TensorFlow optimizer."""
+    tf.random.set_seed(random_seed)
+    for i, (features, labels) in enumerate(zip(X, Y)):
+        with tf.GradientTape() as tape:
+            outputs = model.call(features)
+            loss = loss_fn(outputs, labels)
+        grads = tape.gradient(loss, model.trainable_variables)
+        optimizer.apply_gradients(zip(grads, model.trainable_variables))
+
+
+def _train_edl_embedding_with_optimizer_wrapper(
+    model, opt_keras, X, Y, loss_fn, embed_dims, random_seed
+):
+    """Train model with optimizer wrapper."""
+    tf.random.set_seed(random_seed)
+    optimizer = OptimizerWrapper(opt_keras, None, embed_dims)
+
+    # initialization process related to embedding layer and optimizer wrapper
+    embed_layers = []
+    for layer in model.layers:
+        if isinstance(layer, Embedding):
+            embed_layers.append(layer)
```

Comment:
I remember we have a utility function for this? Reuse `model_helper.find_layer`?

Reply:
Fixed.